### PR TITLE
Release no asg

### PR DESCRIPTION
**Pull request by jmolnar-zscaler**
bug-132511
bug-135151
ztw-51

### CODE REVIEW

File: examples/base_cc_gwlb_zpa/terraform.tfvars [hunk 1]
```diff
@@ -144,26 +144,39 @@
 
 #cross_zone_lb_enabled                      = true
 
+## 15. Gateway loadbalancing hashing algorith. Zscaler recommended default is 2-tuple (source_ip_dest_ip).
```
Comment (vkrishnamurthy-zscaler) [inline examples/base_cc_gwlb_zpa/terraform.tfvars:147]: algorith->algorithm

File: examples/cc_gwlb/terraform.tfvars [hunk 2]
```diff
@@ -105,50 +99,56 @@
...
 
 #cc_callhome_enabled                        = false
 
-
 ## 14. By default, GWLB deployments are configured as zonal. Uncomment if you want to enable cross-zone load balancing
 ##     functionality. Only applicable for gwlb deployment types. (Default: false)
 
 #cross_zone_lb_enabled                      = true
 
+## 15. Gateway loadbalancing hashing algorith. Zscaler recommended default is 2-tuple (source_ip_dest_ip).
```
Comment (vkrishnamurthy-zscaler) [inline examples/cc_gwlb/terraform.tfvars:122]: algorith->agorithm

**Review by vkrishnamurthy-zscaler — COMMENTED** [review]
except for spelling looks good

**Review by vkrishnamurthy-zscaler — COMMENTED** [review]
looks good

File: examples/cc_gwlb/variables.tf [hunk 3]
```diff
@@ -221,6 +221,12 @@ variable "allowed_principals" {
   default     = []
 }
 
+variable "deregistration_delay" {
+  type        = number
+  description = "Amount time for Elastic Load Balancing to wait before changing the state of a deregistering target from draining to unused. The range is 0-3600 seconds."
+  default     = 0
```
Comment (vkrishnamurthy-zscaler) [inline examples/cc_gwlb/variables.tf:248]: I believe you told me about this but I can't remember.  default was 300.  making it 0 means we are not giving any time for existing flows sent to be processed and delivered back.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,10 @@
+## UNRELEASED
+* AWS Provider updated to 4.52.x for all example templates and child modules
+* terraform-zscc-gwlb-aws custom gwlb_name and target_group_name variables added
+* name_prefix variable default string changed from zsdemo to zscc
+* validation length constraint added to name_prefix variable
+* dependency fixes for vpc/subnet data resource selection when vpc was originally created with example templates
+
 ## v0.1.0 (December 15, 2022) 
 * github release refactor from Cloud Connector Portal
 * zsec update for terraform support up to 1.1.9 and aws provider 4.7.x
```
Comment (vkrishnamurthy-zscaler) [inline CHANGELOG.md:15]: 4.7.x -> 4.52.x

File: modules/terraform-zscc-sg-aws/versions.tf [hunk 5]
```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = "~> 4.7.0"
+      version = "~> 4.52.0"
     }
   }
   required_version = ">= 0.13"
```
Comment (vkrishnamurthy-zscaler) [inline modules/terraform-zscc-sg-aws/versions.tf:8]: You seem to follow required_version = ">= 0.13.7, < 2.0.0" but this one doesn't confirm to that

**Review by vkrishnamurthy-zscaler — COMMENTED** [review]
This check needs to be added to base_cc_gwlb_asg and asg_zpa.  I used to hit issues about exceeding 32 characters
validation {
    condition     = length(var.name_prefix) <= 12
    error_message = "Variable name_prefix must be 12 or less characters."
  }

File: CHANGELOG.md (hunk 4 above)
Comment (jmolnar-zscaler) [reply CHANGELOG.md:15]: fixed in commit 13cda2542b0e1ecb28a8e65b6f59cd5309b13dd5

File: modules/terraform-zscc-sg-aws/versions.tf (hunk 5 above)
Comment (jmolnar-zscaler) [reply modules/terraform-zscc-sg-aws/versions.tf]: fixed in commit 13cda2542b0e1ecb28a8e65b6f59cd5309b13dd5

File: examples/cc_gwlb/terraform.tfvars (hunk 2 above)
Comment (jmolnar-zscaler) [reply examples/cc_gwlb/terraform.tfvars]: fixed in commit 13cda2542b0e1ecb28a8e65b6f59cd5309b13dd5

File: examples/base_cc_gwlb_zpa/terraform.tfvars (hunk 1 above)
Comment (jmolnar-zscaler) [reply examples/base_cc_gwlb_zpa/terraform.tfvars]: fixed in commit 13cda2542b0e1ecb28a8e65b6f59cd5309b13dd5

File: examples/base_cc_gwlb/terraform.tfvars [hunk 6]
```diff
@@ -129,26 +129,38 @@
 
 #cross_zone_lb_enabled                      = true
 
+## 15. Gateway loadbalancing hashing algorith. Zscaler recommended default is 2-tuple (source_ip_dest_ip).
```
[the remark this answers is not in the document]
Comment (jmolnar-zscaler) [reply examples/base_cc_gwlb/terraform.tfvars]: fixed in commit 13cda2542b0e1ecb28a8e65b6f59cd5309b13dd5

**Review by vkrishnamurthy-zscaler — COMMENTED** [review]
looks good. Nice work

**Review by vkrishnamurthy-zscaler — APPROVED** [review]
looks good

**Review by vkrishnamurthy-zscaler — APPROVED** [review]

**Pull request merged**